applications/TopologyOptimizationApplication/custom_utilities/structure_response_function_utilities.h

```diff
@@ -0,0 +1,300 @@
+/** Detail class definition.
+
+ */
+
+class StructureResponseFunctionUtilities
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of StructureResponseFunctionUtilities
+    KRATOS_CLASS_POINTER_DEFINITION(StructureResponseFunctionUtilities);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    StructureResponseFunctionUtilities( ModelPart& model_part )
+    : mr_structure_model_part(model_part)
+    {
+    }
+
+    /// Destructor.
+    virtual ~StructureResponseFunctionUtilities()
+    {
+    }
+
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    // ---------------------------------------------------------------------------------------------------------------------------------------------
+    // --------------------------------- COMPUTE STRAIN ENERGY -------------------------------------------------------------------------------------
+    // ---------------------------------------------------------------------------------------------------------------------------------------------
+
+    /// Computes the strain energy as the objective function of the optimization problem.
+    double ComputeStrainEnergy()
+    {
+        BuiltinTimer timer;
+        KRATOS_INFO("[TopOpt]") << "  Start calculating strain energy."<<std::endl;
+
+        double Out = 0.0;
+        double Global_Strain_Energy = 0.0;
+
+        // Loop over all elements to calculate their local objective function and sum it into the global objective function (Global Strain Energy)
+        for( ModelPart::ElementIterator element_i = mr_structure_model_part.ElementsBegin(); element_i!= mr_structure_model_part.ElementsEnd();
+                element_i++ )
+        {
+
+            element_i->Calculate(LOCAL_STRAIN_ENERGY, Out, mr_structure_model_part.GetProcessInfo());
+
+            Global_Strain_Energy += element_i->GetValue(LOCAL_STRAIN_ENERGY);
+            
+        }
+
+        KRATOS_INFO("[TopOpt]") <<  "  Strain energy calculated                  [ spent time =  " << timer.ElapsedSeconds() << " ] " << std::endl;
+
+        // Return this obtained Global Strain Energy value as the objective function of the complete system
+        return Global_Strain_Energy;
+
+        KRATOS_CATCH("");
+    }
+
+    double ComputeVolumeFraction()
+    {
+        KRATOS_TRY;
+
+        BuiltinTimer timer;
+        KRATOS_INFO("[TopOpt]") <<"  Start calculating volume fraction."<<std::endl;
+
+        int number_elements = 0;
+        double Global_Volume_Fraction = 0.0;
+        double elemental_volume = 0.0;
+        double design_variable = 0.0;
+        double Total_volume = 0.0;
+
+
+        // Loop over all elements to obtain their X_PHYS and know how many elements the model has
+        for( ModelPart::ElementIterator element_i = mr_structure_model_part.ElementsBegin(); element_i!= mr_structure_model_part.ElementsEnd();
```

Comment:
Parallel?

Reply:
Parallel utilities please

Reply:
@PhiHo-eng it is documented here: https://github.com/KratosMultiphysics/Kratos/wiki/Parallel-Utilities

Reply:
@armingeiser thank you! But I think in this file I can not really implement a parallel utility. I have seen if I do so, the results are different than before. I think this is based on the fact, that I have to go trough the elements and call the properties in order to get the right result at the end.

Reply:
But I implemented the things from your suggestion in the `structure_adjoint_sensitivity_strategy.h`